apps/blockchain/lib/blockchain/transaction.ex

```diff
@@ -330,11 +331,11 @@ defmodule Blockchain.Transaction do
       iex> Blockchain.Transaction.intrinsic_gas_cost(%Blockchain.Transaction{to: <<>>, init: <<1, 2, 0, 3>>, data: <<>>}, %Block.Header{number: 5_000_000})
       3 * 68 + 4  + 21000
   """
-  @spec intrinsic_gas_cost(t, Header.t()) :: Gas.t()
-  def intrinsic_gas_cost(tx, block_header) do
+  @spec intrinsic_gas_cost(t, EVM.Configuration.t()) :: Gas.t()
+  def intrinsic_gas_cost(tx, config) do
     data_cost = input_data_cost(tx)
-    cc_cost = contract_creation_cost(tx, block_header)
-    data_cost + cc_cost + Gas.g_transaction()
```

Comment:
This used to add the contract creation cost and the transaction cost. I think in practice one of those will always be `0` right? That's why you switched it from adding to an `if`/`else` statement within the function `contract_creation_cost/2`?

Reply:
yes, it's related to [EIP 2](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-2.md). In frontier cost for simple transaction and contract creation transactions were 21_000. Starting from Homestead contract creation transaction cost increased to 53_000